### DISCUSSION TITLE
Will client work synching in ecryptfs home drive?

### DISCUSSION THREAD

**Discussion by johnsidoli** [abraunegg/onedrive#1693] [Q&A] [answered]
I am running LinuxMint and want an ecryptfs setup home drive to sync around 200gb of work files with OneDrive (with all the full support for MS features, sharepoint etc etc).

I am trying to find out if the abraunegg client will work with this setup, but find no mention of ecryptfs. Can anyone confirm?

(Basically, I want my data secure from losing my laptop, but don't mind MS having internal access!)

**Comment by abraunegg** [comment, marked as answer]
@johnsidoli 
I have done a lot of investigations in supporting syncing encrypted data to OneDrive. See: https://github.com/abraunegg/onedrive/issues/1023

The issue with OneDrive and encrypted files is the 400 character limit for the full file path length as stored on OneDrive. What this means is you cannot store on OneDrive the ecryptfs files, but you can store the unencrypted versions on OneDrive.

If your configured 'sync_dir' location is you location of your unencrypted files as presented by ecryptfs, then there should be zero issue as I have tested this configuration extensively with [MyNAS Storage Appliance](http://www.mynas.com.au/)

**Comment by johnsidoli** [comment]
Thank you @abraunegg ... meanwhile, my only question now is whether to use ecryptfs as I understand it is not recommended. But I don't understand what other way I can encrypt my home directory with dual boot linux/windows... it seems to be whole drive or nothing. Anyway, that's not really a question for here :rofl:

**Reply by abraunegg** [reply]
@johnsidoli 
Refer to https://github.com/abraunegg/onedrive/issues/1023 as this has options that other people use such as `cryptomator`
